src/EventListener/ReadListener.php

```diff
@@ -82,7 +83,11 @@ public function onKernelRequest(GetResponseEvent $event): void
         }
 
         if (isset($attributes['collection_operation_name'])) {
-            $request->attributes->set('data', $this->getCollectionData($attributes, $context));
+            try {
+                $request->attributes->set('data', $this->getCollectionData($attributes, $context));
+            } catch (ItemNotFoundException $e) {
+                throw new NotFoundHttpException($e->getMessage(), $e);
+            }
```

Comment:
Does a collection really throws that kind of exception? I doubt it.

Reply:
Sorry, I forgot to change the PHPDoc on the interfaces. This is a new feature that I'm proposing.